src/main/java/htsjdk/samtools/read/Read.java

```diff
@@ -0,0 +1,1438 @@
+     * Subclasses may override to provide more efficient implementations.
+     *
+     * @return number of cigar elements (number + operator) in the cigar string.
+     */
+    default int getCigarLength() {
+        return getCigar().numCigarElements();
+    }
+
+    /**
+     * Sets the {@link Cigar} object describing how the read aligns to the reference.
+     *
+     * @param cigar Cigar object for the read; empty Cigar if there is none.
+     *
+     * @throws IllegalArgumentException if the cigar is {@code null}.
+     */
+    void setCigar(final Cigar cigar);
+
+    /**
+     * Gets the contig name where the read's mate is mapped to. May return null if there is no unique mapping.
+     *
+     * <p>Equivalent to <b>RNEXT</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * @return name of the contig the read's mate is mapped to; {@link ReadConstants#NO_ALIGNMENT_REFERENCE_NAME} if unset (e.g. for unmapped reads).
+     */
+    String getMateContig();
+
+    /**
+     * Gets the 1-based inclusive leftmost position of the sequence remaining after clipping for the read's mate.
+     *
+     * <p>Equivalent to <b>PNEXT</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * @return 1-based start position; {@link ReadConstants#NO_ALIGNMENT_START} if there is no position (e.g. for unmapped read).
+     */
+    int getMateStart();
+
+    /**
+     * Sets the alignment position for the read's mate.
+     *
+     * <p>Equivalent to <b>RNEXT</b> and <b>PNEXT</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * @param contig name of the contig the read's mate is mapped to; {@link ReadConstants#NO_ALIGNMENT_REFERENCE_NAME} if unset (e.g. for unmapped reads).
+     * @param start 1-based start position; {@link ReadConstants#NO_ALIGNMENT_START} if there is no position (e.g. for unmapped read).
+     *
+     * @throws IllegalArgumentException if the contig is {@code null}.
+     */
+    void setMateAlignmentPosition(final String contig, int start);
+
+    /**
+     * Sets the alignment position for the read's mate.
+     *
+     * <p>Equivalent to <b>RNEXT</b> and <b>PNEXT</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Default implementation uses the {@link #setMateAlignmentPosition(String, int)}, ignoring the {@link Locatable#getEnd()}.
+     *
+     * @param position location the read's mate is mapped to.
+     *
+     * @throws IllegalArgumentException if the locatable is {@code null}.
+     */
+    default void setMateAlignmentPosition(final Locatable position) {
+        setMateAlignmentPosition(position.getContig(), position.getStart());
+    }
+
+    /**
+     * Gets the insert size (difference between the 5' end of the read and the 5' end of the mate).
```

Comment:
I used the same documentation as `SAMRecord`. I changed it to the specs documentation, but maybe it would require a change of the name if so. New text:

```java
    /**
     * Gets the signed observed template length. If  all  segments  are  mapped  to  the  same  reference,
     * the unsigned observed template length equals the number of bases from the leftmost mapped base to the
     * rightmost mapped base.  The leftmost segment has a plus sign and the rightmost has a minus sign.
     * The sign of segments in the middle is undefined. It is set as 0 for single-segment template or when the
     * information is unavailable.
     *
     * <p>Equivalent to <b>TLEN</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
     *
     * @return insert size if possible to compute; {@code 0} otherwise.
     */
    int getInferredInsertSize();
```